plasmapy/physics/transport/collisions.py

```diff
@@ -1311,3 +1311,287 @@ def coupling_parameter(T,
+
+    Considering a Maxwellian distribution of "test" electrons colliding with
+    a Maxwellian distribution of "field" ions.
+
+    This result is an electron momentum relaxation rate, and is used in many
+    classical transport expressions. It is equivalent to:
+    * 1/tau_e from ref [1]_ eqn (1) pp. #,
+    * 1/tau_e from ref [2]_ eqn (1) pp. #,
+    * nu_e\i_S from ref [2]_ eqn (1) pp. #,
+
+    Parameters
+    ----------
+    T_e : ~astropy.units.Quantity
+        The electron temperature of the Maxwellian test electrons
+
+    n_e : ~astropy.units.Quantity
+        The number density of the Maxwellian test electrons
+
+    ion_particle: str
+        String signifying a particle type of the field ions, including charge
+        state information.
+
+    V : ~astropy.units.Quantity, optional
+        The relative velocity between particles.  If not provided,
+        thermal velocity is assumed: :math:`\mu V^2 \sim 2 k_B T`
+        where `mu` is the reduced mass.
+
+    coulomb_log : float or dimensionless ~astropy.units.Quantity, optional
+        Option to specify a Coulomb logarithm of the electrons on the ions.
+        If not specified, the Coulomb log will is calculated using the
+        `~plasmapy.physics.transport.Coulomb_logarithm` function.
+
+    coulomb_log_method : string, optional
+        Method used for Coulomb logarithm calculation (see that function
+        for more documentation). Choose from "classical" or "GMS-1" to "GMS-6".
+
+    References
+    ----------
+    .. [1] Braginskii
+
+    .. [2] Formulary
+
+    .. [3] Callen Chapter 2, http://homepages.cae.wisc.edu/~callen/chap2.pdf
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> collision_rate_electron_ion(0.1*u.eV, 1e6/u.m**3, 'p')
+    <Quantity 0.00180172 1 / s>
+    >>> collision_rate_electron_ion(100*u.eV, 1e6/u.m**3, 'p')
+    <Quantity 8.6204672e-08 1 / s>
+    >>> collision_rate_electron_ion(100*u.eV, 1e20/u.m**3, 'p')
+    <Quantity 3936037.8595928 1 / s>
+    >>> collision_rate_electron_ion(100*u.eV, 1e20/u.m**3, 'p', coulomb_log_method = 'GMS-1')
+    <Quantity 3872922.52743562 1 / s>
+    >>> collision_rate_electron_ion(0.1*u.eV, 1e6/u.m**3, 'p', V = c/100)
+    <Quantity 4.41166015e-07 1 / s>
+    >>> collision_rate_electron_ion(100*u.eV, 1e20/u.m**3, 'p', coulomb_log = 20)
+    <Quantity 5812633.74935004 1 / s>
+
+    """
+    T_e = T_e.to(u.K, equivalencies=u.temperature_energy())
+    if V is None:
+        # electron thermal velocity (most probable)
+        V = np.sqrt(2 * k_B * T_e / m_e)
+
+    particles = [ion_particle, 'e-']
+    Z_i = atomic.integer_charge(ion_particle)
+    nu = collision_frequency(T_e,
+                             n_e,
+                             particles,
+                             z_mean=Z_i,
+                             V=V,
+                             method=coulomb_log_method
+                             )
+    coeff = 4 / np.sqrt(np.pi) / 3
```

Comment:
I suspect that this coefficient is going to be different. Under the ion-ion collision version I have `coeff = np.sqrt(8 / np.pi) / 3 / 4` where the last factor of 4 is due to a factor of 2 that shows up in `b_perp` due to reduced mass being different from regular mass (and the frequency goes as `b_perp ** 2`.

Reduced mass in an electron-ion system will be `mu = m_e * m_i / (m_e + m_i)` and if we take `m_e << m_i` it reduces to `mu ~ m_e`. So the thermal velocity looks correct...

There may also be a factor of `Z_i` depending on how the old version of this function was defined in relation to `collision_frequency`.

I think I need to sit down and write these versions out.

Reply:
As for the reduced mass... I had differences between the result and expected values on the order of 0.1%. I just forgot to finish that sentence in the pull request. Anyway, just about right for the electron to proton mass ratio, right? Maybe that's the source of the "error", maybe it's actually correct with the reduced mass?

Reply:
Hm...slight differences of order 0.1% could be caused by interchanging the proton mass with the mass of neutral hydrogen or `H-1`.  I'm not sure where this might have come in here, but it's a problem that I've run into (...or more accurately, a problem I've created a few times :joy_cat:).

Reply:
Here's what I've checked so far
* with respect to reduced mass, `collision_frequency` for the ion-electron case [includes](https://github.com/PlasmaPy/PlasmaPy/blob/28156768b242564265e33b3baf8ae231f1a7c166/plasmapy/physics/transport/collisions.py#L679) has a factor of `m_e / reduced_mass` for `bperp`. The only place where `bperp` is not compensated for is the implicit `bperp` within the `Coulomb_logarithm`. Though this difference will be super super tiny... You can check this by seeing if there is a failure for the case where a custom value of coulomb logarithm is passed.
* particles are passed to `bperp` so Z is correctly calculated in that version. Meaning `bperp` goes as `Z * e * e` (ion-electron interaction). By contrast, `collision_rate_electron_ion` just [uses](https://github.com/PlasmaPy/PlasmaPy/blob/28156768b242564265e33b3baf8ae231f1a7c166/plasmapy/physics/parameters.py#L638) `e ** 2`. So I bet the differences will diverge even further if you do tests with higher Z particles than protons.
* I doubt that it's a difference between protons and hydrogens since the same particles are getting passed to both versions of function. Could there be a difference between `particles = [ion_particle, 'e-']` and `particles = ['e', ion_particle]`? Might be worth testing.

Writing it out I don't see any other differences.

Reply:
Alright, so I just tested this locally and there is indeed a difference when it comes to particles. The difference is due to the negative sign.

For `particles = ['e', ion_particle]` I get:
```
_______________________________________ [doctest] plasmapy.physics.parameters.collision_rate_electron_ion _______________________________________
128     >>> from astropy import units as u
129     >>> collision_rate_electron_ion(0.1*u.eV, 1e6/u.m**3, 'p')
130     <Quantity 0.00180172 1 / s>
131     >>> collision_rate_electron_ion(100*u.eV, 1e6/u.m**3, 'p')
132     <Quantity 8.6204672e-08 1 / s>
133     >>> collision_rate_electron_ion(100*u.eV, 1e20/u.m**3, 'p')
134     <Quantity 3936037.8595928 1 / s>
135     >>> collision_rate_electron_ion(100*u.eV, 1e20/u.m**3, 'p', coulomb_log_method = 'GMS-1')
136     <Quantity 3872922.52743562 1 / s>
137     >>> collision_rate_electron_ion(0.1*u.eV, 1e6/u.m**3, 'p', V = c/100)
Expected:
    <Quantity 4.41166015e-07 1 / s>
Got:
    <Quantity 0.00180172 1 / s>

```

For `particles = ['e-', ion_particle]` I get:

```
_______________________________________ [doctest] plasmapy.physics.parameters.collision_rate_electron_ion _______________________________________
120     .. [1] Braginskii
121 
122     .. [2] Formulary
123 
124     .. [3] Callen Chapter 2, http://homepages.cae.wisc.edu/~callen/chap2.pdf
125 
126     Examples
127     --------
128     >>> from astropy import units as u
129     >>> collision_rate_electron_ion(0.1*u.eV, 1e6/u.m**3, 'p')
Expected:
    <Quantity 0.00180172 1 / s>
Got:
    <Quantity 0.00180368 1 / s>

```

Order doesn't matter, but that negative sign is doing weird things.

Reply:
@StanczakDominik do you want to make these changes or shall I?